tests/thread/enc/errno_tests.cpp

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+#ifdef _PTHREAD_ENC_
+#include "thread.h"
+#endif
+
+#include <openenclave/enclave.h>
+#include <openenclave/internal/print.h>
+#include <openenclave/internal/thread.h>
+#include <openenclave/internal/types.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <algorithm>
+#include "errno.h"
+#include "thread_t.h"
+
+void* enc_malloc(size_t size, int* err)
+{
+    void* ret = malloc(size);
+    *err = errno;
+    return ret;
+}
+
+long enc_strtol(const char* nptr, int base, int* err)
+{
+    long ret = strtol(nptr, NULL, base);
```

Comment:
Both C standard and POSIX require errno to be set for strtol.

Reply:
@anakrish why the case don't get errno from oe strtol on windows?
The follow is the log of the cases running on the windows platform.
[errno_thread1]Error check errno, expected 12, got 12.( this is malloc errno)
[errno_thread2] expected 34, got errno 0.( this is strtol errno)

Reply:
@yanxue-intel Can you use int64_t instead of long? long has different size on Windows and Linux. OE uses ELF format (Linux) on both Linux and Windows. edger8r will generate a warning when these non portable types are used.

Reply:
@anakrish I don't think the errno is affected by output type cast.
I replaced long with int64_t, but the errno was not set either.

Reply:
@yanxue-intel The latest version where `long` has been changed to `int64_t`, the test now passes in Windows. The outstanding failures are in Linux due to timeout.

Reply:
linux oe_thread failure maybe caused by test_tcs_exhaustion, please help to check details in log file, Thanks. [](url)https://oe-jenkins-dev.westeurope.cloudapp.azure.com/blue/rest/organizations/jenkins/pipelines/Bors/branches/staging/runs/273/nodes/71/steps/846/log/?start=0